src/v/compression/include/compression/lz4_decompression_buffers.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/seastarx.h"
+#include "ssx/semaphore.h"
+
+#include <seastar/core/aligned_buffer.hh>
+#include <seastar/core/semaphore.hh>
+
+#define LZ4F_STATIC_LINKING_ONLY
+
+#include <lz4frame.h>
```

Comment:
From reading the comment in https://github.com/lz4/lz4/blob/dev/lib/lz4frame.h#L530 and testing locally it seems the definition has to be explicit, but I can add a guard around it to be safer (I assumed the pragma once in header will already guard it)

Reply:
As an example in their test programs they define this explicitly https://github.com/lz4/lz4/blob/47058197f79c177c597dc20c3aa806d9262bd86c/programs/lz4io.c#L61 but if there is a way to rely on CMake rather than defining it here I can make the change.

Reply:
Yes, reading the documentation, it seems it's the blessed way. 

If we were in control of the build, it would be easy. something like

```
target_compile_definitions(lz4 PUBLIC LZ4F_STATIC_LINKING_ONLY)
```